src/Combinatorics/Matroids/realization_space.jl

```diff
@@ -0,0 +1,810 @@
+
+@attributes mutable struct MatroidRealizationSpace
+  defining_ideal::Union{Ideal,NumFieldOrdIdl}
+  inequations::Vector{RingElem}
+  ambient_ring::Ring
+  realization_matrix::Union{MatElem,Nothing}
+  char::Union{Int,Nothing}
+  q::Union{Int,Nothing}
+  ground_ring::Ring
+  one_realization::Bool
+
+  function MatroidRealizationSpace(
+    I::Union{Ideal,NumFieldOrdIdl},
+    ineqs::Vector{<:RingElem},
+    R::Ring,
+    mat::Union{MatElem,Nothing},
+    char::Union{Int,Nothing},
+    q::Union{Int,Nothing},
+    ground_ring::Ring)
+    return new(I, ineqs, R, mat, char, q, ground_ring, false)
+  end
+end
+
+function Base.show(io::IO, RS::MatroidRealizationSpace)
+  if has_attribute(RS, :is_realizable) && !is_realizable(RS)
+    if RS.char == nothing && RS.q == nothing
+      print(io, "The matroid is not realizable.")
+    else
+      print(io, "The matroid is not realizable over the specified field or characteristic.")
+    end
+  else
+    if RS.one_realization
+      println(io, "One realization is given by")
+    elseif has_attribute(RS, :is_realizable) && is_realizable(RS)
+      println(io, "The realizations are parametrized by")
+    elseif !has_attribute(RS, :is_realizable)
+      println(io, "The realization space is")
+    end
+
+    show(io, MIME("text/plain"), RS.realization_matrix)
+    print(io, "\nin the ", RS.ambient_ring)
+    I = RS.defining_ideal
+    if (I isa NumFieldOrdIdl && I.gen != ZZ(0)) ||
+      (I isa Ideal && !iszero(I))
```

Comment:
This looks to me as if there should be a `iszero(::NumFieldOrdIdl)` method? @thofma does that make sense? Then this could be rewritten as
```suggestion
    if !iszero(I)
```
which I would greatly prefer.

Reply:
I opened an issue https://github.com/thofma/Hecke.jl/issues/1261, but don't count on it to be added before this is merged here.